### ISSUE TITLE
Not well formed URL

### ISSUE THREAD

**Issue by eliemichel** [milktoastlab/SolanaNFTBot#34]
Hello, after working for some time, the bots logs this error at each NFT sale:

```
Error occurred when notifying Discord DiscordAPIError: Invalid Form Body
embeds[0].image.url: Not a well formed URL.
    at RequestHandler.execute (/home/debian/src/SolanaNFTBot/node_modules/discord.js/src/rest/RequestHandler.js:350:13)
    at runMicrotasks (<anonymous>)
    at processTicksAndRejections (node:internal/process/task_queues:96:5)
    at async RequestHandler.push (/home/debian/src/SolanaNFTBot/node_modules/discord.js/src/rest/RequestHandler.js:51:14)
    at async TextChannel.send (/home/debian/src/SolanaNFTBot/node_modules/discord.js/src/structures/interfaces/TextBasedChannel.js:175:15)
    at async notifyDiscordSale (/home/debian/src/SolanaNFTBot/dist/src/lib/discord/notifyDiscordSale.js:80:5)
    at async Object.onTransaction (/home/debian/src/SolanaNFTBot/dist/src/workers/notifyNFTSalesWorker.js:54:29)
    at async fetchWeb3Transactions (/home/debian/src/SolanaNFTBot/dist/src/lib/solana/connection.js:28:79)
    at async Object.execute (/home/debian/src/SolanaNFTBot/dist/src/workers/notifyNFTSalesWorker.js:35:13)
    at async /home/debian/src/SolanaNFTBot/dist/src/workers/initWorkers.js:12:24 {
  method: 'post',
  path: '/channels/888019678230904852/messages',
  code: 50035,
  httpStatus: 400,
  requestData: {
    json: {
      content: undefined,
      tts: false,
      nonce: undefined,
      embeds: [Array],
      components: [Array],
      username: undefined,
      avatar_url: undefined,
      allowed_mentions: undefined,
      flags: undefined,
      message_reference: undefined,
      attachments: undefined,
      sticker_ids: undefined
    },
    files: []
  }
}
```

Do you now how I could debug this?

**Comment by kryptoj** [comment]
This comes from the nft image url. You can double-check that the image URL is working.

**Comment by eliemichel** [comment]
My URL is correct, but contains spaces, I bet you need to urlencode it when sending to Discord

**Comment by kryptoj** [comment]
interesting @eliemichel, can you share the NFT for me to have a look at?

**Comment by eliemichel** [comment]
For instance https://explorer.solana.com/address/FiAi3yRnHRGUXW5CYyCABrvGvQnQnsQsSXQ6jx4XRgCP
Just wrapping the image URL into `encodeURI()` fixed the issue in the end.

**Comment by kryptoj** [comment]
Fixed in v0.13.0
https://github.com/milktoastlab/SolanaNFTBot/releases/tag/v0.130
